rest/bulk_api.go

```diff
@@ -48,6 +48,21 @@ type allDocsRow struct {
 
 // HTTP handler for _all_docs
 func (h *handler) handleAllDocs() error {
+
+	var numDocsPreFilter, numDocsPostFilter uint64
+	defer func() {
+		// increment public all docs stats here
+		if h.privs != adminPrivs {
+			h.db.DbStats.DatabaseStats.NumPublicAllDocsRequests.Add(1)
+			if numDocsPreFilter > 0 {
+				h.db.DbStats.DatabaseStats.NumDocsPreFilterPublicAllDocs.Add(int64(numDocsPreFilter))
+			}
+			if numDocsPostFilter > 0 {
+				h.db.DbStats.DatabaseStats.NumDocsPostFilterPublicAllDocs.Add(int64(numDocsPostFilter))
+			}
+		}
+	}()
```

Comment:
Pull the conditional up out so we avoid the defer cost - `h.privs` won't change mid-request, so we know up-front whether to do this or not.

```suggestion
	// public _all_docs stats
	if h.privs != adminPrivs {
		defer func() {
			h.db.DbStats.DatabaseStats.NumPublicAllDocsRequests.Add(1)
			if numDocsPreFilter > 0 {
				h.db.DbStats.DatabaseStats.NumDocsPreFilterPublicAllDocs.Add(int64(numDocsPreFilter))
			}
			if numDocsPostFilter > 0 {
				h.db.DbStats.DatabaseStats.NumDocsPostFilterPublicAllDocs.Add(int64(numDocsPostFilter))
			}
		}()
	}
```